### PR TITLE
Add navigation plugin

### PR DESCRIPTION
**Pull request by HidenoriMatsubayashi**
I added flutter/navigation plugin support. But it isn't called from anywhere at least now. Also, probably it would be used for the back button only.

### CODE REVIEW

**Comment by HidenoriMatsubayashi** [comment]
@taka-okayama Can you review the PR?

File: src/flutter/shell/platform/linux_embedded/plugin/navigation_plugin.cc [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  LINUXES_LOG(DEBUG) << "PushRoute = " << route;
+
+  auto args = std::make_unique<rapidjson::Document>(rapidjson::kObjectType);
+  auto& allocator = args->GetAllocator();
+  args->Parse("\"" + route + "\"");
+  if (args->HasParseError()) {
+    LINUXES_LOG(ERROR) << "Failed to parse the route: " << route;
+    return ;
+  }
+  channel_->InvokeMethod(kSetInitialRouteMethod, std::move(args));
```
Comment (taka-okayama) [inline src/flutter/shell/platform/linux_embedded/plugin/navigation_plugin.cc:47]: kPushRouteMethod?

Comment (HidenoriMatsubayashi) [reply src/flutter/shell/platform/linux_embedded/plugin/navigation_plugin.cc]: I fixed it.

**Review by taka-okayama — APPROVED** [review]
ok.

**Pull request merged**